### ISSUE TITLE
Complete notifications deemed to be signals

### ISSUE THREAD

**Issue by cartant** [ReactiveX/rxjs#5838]
## Bug Report

With numerous operators, the `next` and `complete` notifications received from notifier (or selector) observables are treated in the same way. As commented [here](https://github.com/ReactiveX/rxjs/blob/8ef4fee2b5f7fb68426661ffcc47b50ab81b145e/src/internal/operators/delayWhen.ts#L141-L142), that is arguably incorrect behaviour (at the very least, IIRC, this behaviour is inconsistent throughout the API):

```ts
// TODO(benlesh): I'm inclined to say this is _incorrect_ behavior.
// A completion should not be a notification. Note the deprecation above
```

AFAICT, the following operators effect this incorrect behaviour:

- [x] [`audit`](https://github.com/ReactiveX/rxjs/blob/8ef4fee2b5f7fb68426661ffcc47b50ab81b145e/src/internal/operators/audit.ts#L80)
- [x] [`bufferToggle`](https://github.com/ReactiveX/rxjs/blob/8ef4fee2b5f7fb68426661ffcc47b50ab81b145e/src/internal/operators/bufferToggle.ts#L83)
- [x] [`bufferWhen`](https://github.com/ReactiveX/rxjs/blob/8ef4fee2b5f7fb68426661ffcc47b50ab81b145e/src/internal/operators/bufferWhen.ts#L72)
- [x] [`debounce`](https://github.com/ReactiveX/rxjs/blob/8ef4fee2b5f7fb68426661ffcc47b50ab81b145e/src/internal/operators/debounce.ts#L100)
- [x] [`delayWhen`](https://github.com/ReactiveX/rxjs/blob/8ef4fee2b5f7fb68426661ffcc47b50ab81b145e/src/internal/operators/delayWhen.ts#L136-L144) (Nope, `delayWhen` is fine. It seems to have been refactored in another PR.)
- [x] [`sample`](https://github.com/ReactiveX/rxjs/blob/8ef4fee2b5f7fb68426661ffcc47b50ab81b145e/src/internal/operators/sample.ts#L64)
- [x] [`throttle`](https://github.com/ReactiveX/rxjs/blob/8ef4fee2b5f7fb68426661ffcc47b50ab81b145e/src/internal/operators/throttle.ts#L84)
- [x] [`windowToggle`](https://github.com/ReactiveX/rxjs/blob/8ef4fee2b5f7fb68426661ffcc47b50ab81b145e/src/internal/operators/windowToggle.ts#L95)

For more context, this was checked off in the list of non-type-related things that need to be done before v7 can be released, but it is not that case that all of these behaviours have been corrected. See #5372

**Comment by cartant** [comment]
Removed the 'Agenda Item' label. In last week's meeting it was decided that these operators should be changed so that the behaviour is consistent and `complete` is not treated as a signal.
